drivers/can/can_mcux_flexcan.c

```diff
@@ -4,7 +4,11 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifdef CONFIG_DT_HAS_NXP_CANFD_FLEXCAN_ENABLED
+#define DT_DRV_COMPAT nxp_canfd_flexcan
+#else
 #define DT_DRV_COMPAT nxp_kinetis_flexcan
+#endif
```

Comment:
This won't work. These SoCs have both classic FlexCAN controllers and FlexCAN-FD controllers. We need to be able to use both at the same time. Your changes to the classic FlexCAN driver only ever allows using one type or the other.

Given that this is a completely different hardware IP (which the NXP MCUX HAL wraps in a similar API as the classic FlexCAN) - and the amount of changes to the existing, classic FlexCAN driver - I think I would go for a completely new driver (with core FlexCAN Zephyr shim functions split into a common C file).

Reply:
For now, I have split them into 2 different files. One pertaining to the classic FlexCAN and the other for the FlexCAN-FD. Let me know if this is what we are looking for. I'll refactor the common functions across the files in the next commit.

Reply:
Splitting the common functions doesn't seem feasible, since the types in the functions are more or less different.